tests/ClientTest.php

```diff
@@ -0,0 +1,292 @@
+        self::expectExceptionMessage(
+            'Invalid parameters when creating client: Value "invalid url" is invalid: This value is not a valid URL.'
+            . "\n" . 'Value "" is invalid: This value should not be blank.' . "\n"
+        );
+        new EventsClient(new NullLogger(), 'invalid url', '');
+    }
+
+    private function getPostEventData(): PostEventRequest
+    {
+        return new PostEventRequest(
+            'job_failed',
+            new FailedJobEventData(
+                'job failed',
+                new JobData('my-project', '123', 'http://someUrl', '2020-01-02', '2020-01-01', 'my-orchestration')
+            )
+        );
+    }
+
+    public function testClientRequestResponse(): void
+    {
+        $mock = new MockHandler([
+            new Response(
+                202,
+                ['Content-Type' => 'application/json'],
+                null
+            ),
+        ]);
+        // Add the history middleware to the handler stack.
+        $requestHistory = [];
+        $history = Middleware::history($requestHistory);
+        $stack = HandlerStack::create($mock);
+        $stack->push($history);
+        $client = $this->getClient(['handler' => $stack]);
+        $client->postEvent($this->getPostEventData());
+        /** @var Request $request */
+        $request = $requestHistory[0]['request'];
+        self::assertEquals('http://example.com/events/job_failed', $request->getUri()->__toString());
+        self::assertEquals('POST', $request->getMethod());
+        self::assertEquals('testToken', $request->getHeader('X-Kbc-ManageApiToken')[0]);
+        self::assertEquals('Notification PHP Client', $request->getHeader('User-Agent')[0]);
+        self::assertEquals('application/json', $request->getHeader('Content-type')[0]);
+    }
+
+    public function testLogger(): void
+    {
+        $mock = new MockHandler([
+            new Response(
+                200,
+                ['Content-Type' => 'application/json'],
+                '{
+                    "id": "683194249",
+                    "runId": "683194249",
+                    "status": "created",
+                    "desiredStatus": "processing",
+                    "mode": "run",
+                    "component": "keboola.ex-db-snowflake",
+                    "config": "123",
+                    "configRow": null,
+                    "tag": null,
+                    "createdTime": "2021-03-04T21:59:49+00:00"
+                }'
+            ),
+        ]);
+        // Add the history middleware to the handler stack.
+        $requestHistory = [];
+        $history = Middleware::history($requestHistory);
+        $stack = HandlerStack::create($mock);
+        $stack->push($history);
+        $logger = new TestLogger();
+        $client = $this->getClient(['handler' => $stack, 'logger' => $logger, 'userAgent' => 'test agent']);
+        $client->postEvent($this->getPostEventData());
+        /** @var Request $request */
```

Comment:
nebyl by lepsi `assertInstanceOf`?

Reply:
no našeptávání v PHPStormu mi to nespraví, mělo by?

Reply:
Kdyz je vse jak ma, tak tomu rozumi

![Screenshot from 2021-07-06 15-45-52](https://user-images.githubusercontent.com/271753/124611156-74f7ce00-de71-11eb-9c13-2c56846f7701.png)

Reply:
No smůla no https://github.com/guzzle/guzzle/blob/de6f1e58e735754b888649495ed4cb9ae3b19589/src/Middleware.php#L85